nengo/params.py

```diff
@@ -187,18 +233,37 @@ def __set__(self, instance, ndarray):
             ndarray = self.validate(instance, ndarray)
         self.data[instance] = ndarray
 
-    def validate(self, instance, ndarray):
-        ndim = len(self.shape)
-        try:
-            ndarray = np.asarray(ndarray, dtype=np.float64)
-        except TypeError:
-            raise ValueError("Must be a float NumPy array (got type '%s')"
-                             % ndarray.__class__.__name__)
-
-        if ndarray.ndim != ndim:
+    def validate(self, instance, ndarray):  # noqa: C901
+        if isinstance(ndarray, np.ndarray):
+            ndarray = ndarray.view()
```

Comment:
This makes it possible to have non-float arrays. Is this intentional? It seems useful some places, but dangerous in others.